src/Metadata/TargetsMetadata.php

```diff
@@ -42,6 +44,28 @@ public static function createFromJson(string $json, string $roleName = null): st
         return $newMetadata;
     }
 
+    /**
+     * Validates that delegated role names are unique.
+     *
+     * @todo Use Symfony's Unique constraint for this when at least Symfony
+     *   6.1 is required in https://github.com/php-tuf/php-tuf/issues/317.
+     *
+     * @param mixed $delegations
+     *   The value to be validated.
+     * @param \Symfony\Component\Validator\Context\ExecutionContextInterface $context
+     *   The validation context.
+     */
+    public static function validateDelegatedRoles(mixed $delegations, ExecutionContextInterface $context): void
+    {
+        if (!is_array($delegations)) {
```

Comment:
Would it possible to add the validation on the `'roles' => new All([` level so you would not have to check if this was set?

Reply:
According to https://symfony.com/doc/current/reference/constraints/All.html, that's not what `All` does.

The reason I had to add this check, for posterity, is because type hinting `$delegations` as an array causes some tests to fail -- specifically, `MetadataBaseTest::testInvalidFields()`, which specifically passes the wrong data type in, so as to confirm we're testing data types. That test coverage is valuable, but fundamentally incompatible with adding a type hint here, because the test will start failing with `TypeError`.